File: Examples/AspNetMvcVB/Controllers/DocumentViewerController.ClientEvents.vb
Imports GleamTech.DocumentUltimate.AspNet.UI
Imports GleamTech.Examples

Namespace Controllers
    Partial Public Class  DocumentViewerController
        Inherits Controller

        <AcceptVerbs(HttpVerbs.Get Or HttpVerbs.Post)>
        Public Function ClientEvents(fileSelector As String) As ActionResult
            Dim exampleFileSelector = New ExampleFileSelector() With {
                .Id = "exampleFileSelector",
                .InitialFile = "Default.pdf"
            }
            ViewBag.ExampleFileSelector = exampleFileSelector

            Dim documentViewer = New DocumentViewer() With {
                .Width = 960,
                .Height = 720,
                .Resizable = True,
                .Document = exampleFileSelector.SelectedFile.ToString(),
                .ClientEvents = New DocumentViewerClientEvents() With {
                    .Loaded = "documentViewerLoaded",
                    .Failed = "documentViewerFailed",
                    .DocumentLoaded = "documentViewerDocumentLoaded",
                    .PageChanged = "documentViewerPageChanged",
                    .PageRendered = "documentViewerPageRendered",
                    .RotationChanged = "documentViewerRotationChanged",
                    .Downloading = "documentViewerDownloading",
                    .Printing = "documentViewerPrinting",
                    .Printed = "documentViewerPrinted",
                    .TextSelected = "documentViewerTextSelected",
                    .TextCopied = "documentViewerTextCopied"
                }
            }

            Return View(documentViewer)
        End Function

    End Class
End Namespace

File: Examples/AspNetMvcCS/Controllers/DocumentViewerController.ClientEvents.cs
using System.Web.Mvc;
using GleamTech.DocumentUltimate.AspNet.UI;
using GleamTech.Examples;

namespace GleamTech.DocumentUltimateExamples.AspNetMvcCS.Controllers
{
    public partial class DocumentViewerController
    {
        [AcceptVerbs(HttpVerbs.Get | HttpVerbs.Post)]
        public ActionResult ClientEvents()
        {
            var exampleFileSelector = ViewBag.ExampleFileSelector = new ExampleFileSelector
            {
                Id = "exampleFileSelector",
                InitialFile = "Default.pdf"
            };

            var documentViewer = new DocumentViewer
            {
                Width = 960,
                Height = 720,
                Resizable = true,
                Document = exampleFileSelector.SelectedFile.ToString(),
                ClientEvents = new DocumentViewerClientEvents {
                    Loaded = "documentViewerLoaded",
                    Failed = "documentViewerFailed",
                    DocumentLoaded = "documentViewerDocumentLoaded",
                    PageChanged = "documentViewerPageChanged",
                    PageRendered = "documentViewerPageRendered",
                    RotationChanged = "documentViewerRotationChanged",
                    Downloading = "documentViewerDownloading",
                    Printing = "documentViewerPrinting",
                    Printed = "documentViewerPrinted",
                    TextSelected = "documentViewerTextSelected",
                    TextCopied = "documentViewerTextCopied"
                }
            };

            return View(documentViewer);
        }
    }
}
